### PR TITLE
[App Service] BREAKING CHANGE: `az appservice ase create`: Update the default App Service Environment to V3

### PR DESCRIPTION
**Pull request by seligj95**
**Related command**

az appservice ase create

**Description**

Currently default ASE kind is set to ASEv2. ASEv2 is retiring in August 2024 and creates of ASEv2 will be blocked starting in January. So we need the default to be updated to ASEv3 to prevent errors and ensure customers are getting onto the latest version.

**Testing Guide**

az appservice ase create -g <rg> -n <asename> --vnet-name <vnet> --subnet <subnet> : use this to create an ASEv3 since the default is now set to ASEv3. You no longer need to include the kind parameter for ASEv3 creates.

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

**Review by madsd — APPROVED** [review]
LGTM

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by seligj95** [comment]
@yonzhan CI issues fixed. Ready for review.

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def show_appserviceenvironment(cmd, name, resource_group_name=None):
     return ase_client.get(resource_group_name, name)
 
 
-def create_appserviceenvironment_arm(cmd, resource_group_name, name, subnet, kind='ASEv2',
+def create_appserviceenvironment_arm(cmd, resource_group_name, name, subnet, kind='ASEv3',
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py:53]: May I ask if this change of default behavior will cause a breaking change? If so, please add the `BREAKING CHANGE` tag in the PR title

**Comment by seligj95** [comment]
@zhoxing-ms This is ready for review. Please let me know if anything else is needed. Thanks!

**Comment by zhoxing-ms** [comment]
Could you add some scenario tests to test the PR change?

**Comment by seligj95** [comment]
> Could you add some scenario tests to test the PR change?

The existing scenario test has been updated to match the functionality update in the PR. Not sure what other scenario test is needed. Can you please share more info?

**Comment by zhoxing-ms** [comment]
@seligj95 What I mean is that it's not just modifying the mock test, it needs to modify the scenario test and re-record the yaml file in live mode

**Comment by seligj95** [comment]
somehow managed to completely mess up this PR. started a new one: https://github.com/Azure/azure-cli/pull/27761

**Pull request closed without merging**